lib/_http_client.js

```diff
@@ -330,10 +334,12 @@ ClientRequest.prototype.abort = function abort() {
   // the request queue through handling in onSocket.
   if (this.socket) {
     // in-progress
-    this.socket.destroy();
+    this.socket.destroy(error);
+  } else {
```

Comment:
This else branch was not needed becase the destroy logic was handled when a socket was assigned to the request. Any reason for adding it now?

Reply:
Because of the TODO. We need to forward the `error` to that logic.

Reply:
If acceptable I'd like to fix that as a separate PR...

Reply:
Updated. Should be more clear now.

Reply:
Ok, yes it is.

Reply:
Fixed.